### PR TITLE
fix fuzz target

### PR DESCRIPTION
**Pull request by kjk**
I was trying to build the fuzz target of `promql` and noticed it doesn't compile anymore.

This change fixes it.

I looked briefly but couldn't figure out if `ParseStmts` is gone or is available under a different name, so for now I've deleted it.

### CODE REVIEW

**Comment by yevgenypats** [comment]
@fabxc @msiebuhr @juliusv

File: promql/fuzz.go [hunk 1]
```diff
@@ -50,11 +54,19 @@ const (
 // Note that his is not the parser for the text-based exposition-format; that
 // lives in github.com/prometheus/client_golang/text.
 func FuzzParseMetric(in []byte) int {
-	p := textparse.New(in)
-	for p.Next() {
+	p := textparse.New(in, "")
```
Comment (juliusv) [inline promql/fuzz.go]: Would be good to also test the OpenMetrics parser in addition to the current Prometheus metrics parser. Maybe just do the same fuzzing a second time with a `application/openmetrics-text` content type parameter?

File: promql/fuzz.go [hunk 2]
```diff
@@ -80,13 +92,3 @@ func FuzzParseExpr(in []byte) int {
 
 	return fuzzMeh
 }
-
-// Fuzz the parser.
-func FuzzParseStmts(in []byte) int {
```
Comment (juliusv) [inline promql/fuzz.go]: Could you also remove the reference to this in the comments at the beginning of the file:

    // Repeat for ParseMetricSeletion, ParseExpr and ParseStmt.

Ok, there's more mis-named stuff in there, maybe fix that as well :)

**Comment by msiebuhr** [comment]
Related to this, I recently ran into https://fuzzbuzz.io/ that will fuzz (some) OSS projects continuously.

Perhaps this is something Prometheus should sign up for?

**Comment by yevgenypats** [comment]
Hi Morten,

You could use fuzzbuzz ass well. We fuzz OSS projects for free and we help
contribute the PR. Is there any reason you do not want to use us?

You can see all the OSS projects we fuzz on our site.

**Comment by yevgenypats** [comment]
@msiebuhr 
some differences between fuzzbuzz and [fuzzit.dev](https://fuzzit.dev)
* Our cli is open-source https://github.com/fuzzitdev/fuzzit which is think is a best practice
* we provide regression fuzz testing during PRs that will help catch bugs early on in a PR
* IMHO we have currently a bit of wider adoption is OSS porjects - you can see on our size.
* In the end I guess you should try both and decide what best for the project.

Please let me know and I'll be happy to contribute a PR once the fuzz targets works.

Cheers,
Yevgeny (Founder @fuzzitdev)

**Comment by juliusv** [comment]
Hi both, I haven't heard of either of those services, but fuzzbuzz looks interesting. @yevgenypats would the main benefit of putting this into a separate service rather than our normal CI with go-fuzz be that fuzzing may take some time, so it's generally good to separate it out? Plus, then you have a specialized UI around fuzz test results. I know little about fuzzing technology details, but I would be interested to see what an integration for Prometheus would look like.

**Comment by juliusv** [comment]
@yevgenypats Oh sorry, misunderstood you and thought you were part of fuzzbuzz, not fuzzit, so just take my same comment for fuzzit :)

**Comment by yevgenypats** [comment]
@juliusv lol the names are indeed confusing:) So first if you won't to deep dive a bit on how continuous fuzzing workflow looks like you can take a look at our blog https://fuzzit.dev/blog. but TLDR:

1) The fuzzing is indeed infinite process and this is not blokcing the CI. every push to master new fuzzers are built and updated to fuzzit so they can continue running and will alert via email/slack/other channel if new crashes are found.
2) Fuzzing generates thousands of useful test-cases as well as crashes. We make use of those test-cases and run them through the fuzzers on every PR. This usually takes seconds (minutes) and prevent from old crashes propagating again or new bugs being introduced. This saved our users literally hundreds of bugs from being introduced.

I hope it was a good TLDR but more info on our blog and feel free to ask me more questions. also you can join our slack:) https://slack.fuzzit.dev

File: promql/fuzz.go (hunk 1 above)
Comment (kjk) [reply promql/fuzz.go]: done

File: promql/fuzz.go (hunk 2 above)
Comment (kjk) [reply promql/fuzz.go]: done

File: promql/fuzz.go [hunk 3]
```diff
@@ -45,22 +50,38 @@ const (
...
+	if err == nil {
 		return fuzzInteresting
 	}
 
 	return fuzzMeh
 }
 
+// Fuzz the metric parser.
+//
+// Note that his is not the parser for the text-based exposition-format; that
```
Comment (juliusv) [inline promql/fuzz.go]: his -> this

**Comment by juliusv** [comment]
@kjk Looks great besides the nit, thanks!

CI is still failing due to missing / wrong gofmt:

```
>> checking code style
gofmt checking failed!
diff -u ./promql/fuzz.go.orig ./promql/fuzz.go
--- ./promql/fuzz.go.orig	2019-08-09 17:24:58.072665453 +0000
+++ ./promql/fuzz.go	2019-08-09 17:24:58.072665453 +0000
@@ -38,7 +38,6 @@
 //
 // Repeat for FuzzParseOpenMetric, FuzzParseMetricSelector and FuzzParseExpr.
 
-
 // Tuning which value is returned from Fuzz*-functions has a strong influence
 // on how quick the fuzzer converges on "interesting" cases. At least try
 // switching between fuzzMeh (= included in corpus, but not a priority) and

Please ensure you are using go version go1.12.7 linux/amd64 for formatting code.
make: *** [Makefile.common:117: common-style] Error 1
Exited with code 2
```

Also, could you add a Signed-off-by line to make the DCO check pass (see https://probot.github.io/apps/dco/)?

**Comment by juliusv** [comment]
@yevgenypats I don't know what the rest of the Prometheus team thinks, but that sounds good - would like to take you up on your offer to send a PR about that :)

**Comment by yevgenypats** [comment]
@juliusv sounds good! once we get this passing the CI and merged we will create a PR for [Fuzzit](https://fuzzit.dev) integration:)

**Comment by kjk** [comment]
@juliusv  I'll open another PR. I followed DCO instructions but `git push --force-with-lease origin fix-fuzz` was apparently not the right thing to do.

The concurrent changes on master made this a mess of rebasing.

**Comment by juliusv** [comment]
Oh hum weird, yeah, something's off here now :)

**Comment by kjk** [comment]
Replaced by https://github.com/prometheus/prometheus/pull/5851

**Pull request closed without merging**